### PR TITLE
support tf.IndexedSlices gradient for dense param

### PR DESCRIPTION
**Pull request by QiJune**
Current optimizer of Go PS could not handle native keras.embedding layer. It will generate a `tf.IndexedSlice` gradient, but the parameter is in NonEmbeddingParam.

This PR fix it.

### CODE REVIEW

File: elasticdl/docker/Dockerfile.dev [hunk 1]
```diff
@@ -37,23 +57,3 @@ RUN cd /tmp/elasticdl && make -f elasticdl/Makefile && \
     export GO111MODULE=on && \
     go mod init elasticdl.org/elasticdl && \
     go install ./... && cd / && rm -rf /tmp/elasticdl
-
-# Copy the data generation package to /var and run them from there.
```
Comment (QiJune) [inline elasticdl/docker/Dockerfile.dev:41]: In my local test environment, I find that I have to build the elasticdl:dev if I modify Go code.
Just move commands not related with Go up, since it will be cached. This will reduce rebuild time.

Comment (skydoorkai) [reply elasticdl/docker/Dockerfile.dev:41]: Better to put this change is a separate PR.
Looks fine to me otherwise.

**Review by skydoorkai — APPROVED** [review]

**Comment by QiJune** [comment]
The unit test is under developing. Currently, please do not merge this PR.
